src/mtcp/mtcp_restart.c

```diff
@@ -239,9 +239,11 @@ MTCP_PRINTF("Attach for debugging.");
       mtcp_abort();
     }
     // This assumes that the MTCP header signature is unique.
-    do {
-      mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
-    } while (mtcp_strcmp(mtcpHdr.signature, MTCP_SIGNATURE) != 0);
+    mtcp_readfile(rinfo.fd, &mtcpHdr, sizeof mtcpHdr);
```

Comment:
I am not sure if I understand this patch.  This 'else' branch is taken only by readmtcp in which case, there might still be DMTCP data (ProcessInfo for parent-child maps, etc.) that needs to be read and ignored. With this patch, readmtcp would always fail because the first page of the ckpt image is always going to be related to the ProcessInfo metadata.

I do understand the problem you are trying to address though. We should instead check the return value of mtcp_readfile and abort if it's unable to read `sizeof mtcpHdr` data.